### PR TITLE
added verifyAuthToken.ts, but needs testing

### PR DESCRIPTION
**Pull request by zbrandon98**

### CODE REVIEW

**Comment by zbrandon98** [comment]
Just noticed it failed lint check. Will fix when I get to campus

File: app/(api)/_datalib/auth/verifyAuthToken.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+export async function verifyAuthToken(token: string) {
+  try {
+    const verifiedPayload = jwt.verify(token, process.env.JWT_SECRET as Secret);
+
+    if (!verifiedPayload) {
+      throw new HttpError('Unauthorized. Invalid token.');
+    }
+
+    return { ok: true, body: verifiedPayload };
```
Comment (Austin2Shih) [inline app/(api)/_datalib/auth/verifyAuthToken.ts:12]: return `{ ok: true, body: verifiedPayload, error: null }`
It's nice to keep a similar structure for both error and success.

for error: return `{ ok: true, body: null, error: error_stuff }`

**Review by Austin2Shih — CHANGES_REQUESTED** [review]
Just a small change and I'll merge after!

File: app/(api)/_datalib/auth/verifyAuthToken.ts (hunk 1 above)
Comment (zbrandon98) [reply app/(api)/_datalib/auth/verifyAuthToken.ts]: Updated changes!

**Review by Austin2Shih — APPROVED** [review]
LGTM!

**Pull request merged**